File: app/src/main/java/com/msaggik/algorithmsdatastructures/java/graph/graph_tree/data_structures/QueueImpl.java
package com.msaggik.algorithmsdatastructures.java.graph.graph_tree.data_structures;

import java.util.LinkedList;

public class QueueImpl<D> implements Queue<D> {

    private LinkedList<D> list = new LinkedList<>();

    @Override
    public void offer(D data) {
        list.addLast(data);
    }

    @Override
    public D poll() {
        return list.removeFirst();
    }

    @Override
    public D peek() {
        return list.getFirst();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }
}

File: app/src/main/java/com/msaggik/algorithmsdatastructures/kotlin/graph/graph_tree/data_structures/QueueImpl.kt
package com.msaggik.algorithmsdatastructures.kotlin.graph.graph_tree.data_structures

import java.util.LinkedList

class QueueImpl<D> : Queue<D> {
    private val list = LinkedList<D>()

    override fun offer(data: D) {
        list.add(data)
    }

    override fun poll(): D {
        return list.removeFirst()
    }

    override fun peek(): D {
        return list.first
    }

    override val isEmpty: Boolean
        get() = list.isEmpty()
}